rtengine/demosaic_algos.cc

```diff
@@ -682,6 +682,41 @@ void RawImageSource::ppg_demosaic()
     free (image);
 }
 
+void RawImageSource::border_interpolate(unsigned int border, float (*image)[3], unsigned int start, unsigned int end)
+{
+    unsigned row, col, y, x, f;
+    float sum[8];
+    unsigned int width = W, height = H;
+    unsigned int colors = 3;
+
+    if (end == 0 ) {
+        end = H;
+    }
+
+    for (row = start; row < end; row++)
+        for (col = 0; col < width; col++) {
+            if (col == border && row >= border && row < height - border) {
```

Comment:
Maybe there should be some checks for `border` relative to `start` and `height` when entering this function.